Source/Common/DateTimeToIso8601DateValuesJsonConverter.cs

```diff
@@ -17,26 +18,31 @@ public class DateTimeToIso8601DateValuesJsonConverter : ValuesJsonConverter
         /// </summary>
         /// <param name="writer">The JSON writer.</param>
         /// <param name="value">The value to write.</param>
-        /// <param name="serializer">The JSON serializer.</param>
-        public override void WriteObject(JsonWriter writer, object? value, JsonSerializer serializer)
+        /// <param name="options">The JSON serializer options.</param>
+        public override void WriteObject(Utf8JsonWriter writer, object? value, JsonSerializerOptions options)
         {
             if (writer is null)
             {
                 throw new ArgumentNullException(nameof(writer));
             }
 
-            if (serializer is null)
+            if (value is null)
             {
-                throw new ArgumentNullException(nameof(serializer));
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            if (options is null)
+            {
+                throw new ArgumentNullException(nameof(options));
             }
 
             if (value is DateTime dateTimeType)
```

Comment:
Now that we have the `DateOnly` type coming, we'll need to rethink this strategy.